pages/en/cookbook/grafting.mdx

```diff
@@ -0,0 +1,215 @@
+---
+title: Grafting on Existing Subgraphs
+---
+
+In this guide, you will learn how to build and deploy new subgraphs by grafting existing subgraphs.
+
+## What is Grafting?
+
+Grafting reuses the data from an existing subgraph and starts indexing it at a later block. This is useful during development to get past simple errors in the mappings quickly or to temporarily get an existing subgraph working again after it has failed. Also, it can be used when adding a feature to a subgraph that takes long to index from scratch.
+
+The grafted subgraph can use a GraphQL schema that is not identical to the one of the base subgraph, but merely compatible with it. It has to be a valid subgraph schema in its own right, but may deviate from the base subgraph's schema in the following ways:
+
+- It adds or removes entity types
+- It removes attributes from entity types
+- It adds nullable attributes to entity types
+- It turns non-nullable attributes into nullable attributes
+- It adds values to enums
+- It adds or removes interfaces
+- It changes for which entity types an interface is implemented
+
+For more information, you can check:
+
+- [Grafting](https://thegraph.com/docs/en/developing/creating-a-subgraph/#grafting-onto-existing-subgraphs)
+
+In this tutorial, we will be covering a basic usecase. We will replace an existing contract with an identical contract (with a new address). Then, graft the existing subgraph onto the new contract.
+
+## Building an Existing Subgraph
+
+To be able to build and deploy the existing subgraph used in this tutorial, you need to follow the instructions found here:
+
+1. `@graphprotocol/graph-cli`- This is a command-line tool for building and deploying subgraphs. [Click here](https://www.npmjs.com/package/@graphprotocol/graph-cli) to download using `npm`.
+
+Once these are installed, you can start building the subgraph with the instructions found here:
+
+- [Subgraph example repo](https://github.com/t-proctor/grafting-tutorial)
+
+> Note: The contract was taken from the following [Hackathon Starterkit](https://github.com/schmidsi/hackathon-starterkit).
+
+## Subgraph's Components
+
+There are three components of a subgraph:
```

Comment:
I don't know if we want to restate this information here, I think we should be able to redirect folks to the existing quick start